internal/controller/isbservicerollout_controller_test.go

```diff
@@ -18,73 +18,172 @@ package controller
-		}
-		isbservicerollout := &apiv1.ISBServiceRollout{}
-
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind ISBServiceRollout")
-			err := k8sClient.Get(ctx, typeNamespacedName, isbservicerollout)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &apiv1.ISBServiceRollout{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-					Spec: apiv1.ISBServiceRolloutSpec{
-						InterStepBufferService: runtime.RawExtension{
-							Raw: []byte(`{"field":"val"}`),
-						},
-					},
+	const (
+		namespace             = "default"
+		isbServiceRolloutName = "isbservicerollout-test"
+		timeout               = 10 * time.Second
+		interval              = 250 * time.Millisecond
+	)
+
+	ctx := context.Background()
+
+	isbsSpec := numaflowv1.InterStepBufferServiceSpec{
+		Redis: &numaflowv1.RedisBufferService{},
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: "latest",
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &numaflowv1.DefaultVolumeSize,
+			},
+		},
+	}
+
+	isbsSpecRaw, err := json.Marshal(isbsSpec)
+	Expect(err).ToNot(HaveOccurred())
+
+	isbServiceRollout := &apiv1.ISBServiceRollout{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      isbServiceRolloutName,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: runtime.RawExtension{
+				Raw: isbsSpecRaw,
+			},
+		},
+	}
+
+	resourceLookupKey := types.NamespacedName{Name: isbServiceRolloutName, Namespace: namespace}
+
+	Context("When applying a ISBServiceRollout spec", func() {
+		It("Should create the ISBServiceRollout if it does not exist", func() {
+			Expect(k8sClient.Create(ctx, isbServiceRollout)).Should(Succeed())
+
+			createdResource := &apiv1.ISBServiceRollout{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, resourceLookupKey, createdResource)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("Verifying the content of the ISBServiceRollout spec field")
+			createdInterStepBufferServiceSpec := numaflowv1.InterStepBufferService{}
+			Expect(json.Unmarshal(createdResource.Spec.InterStepBufferService.Raw, &createdInterStepBufferServiceSpec)).ToNot(HaveOccurred())
+			Expect(createdResource.Spec).Should(Equal(isbServiceRollout.Spec))
+		})
+
+		It("Should have created an ISBServiceRollout", func() {
+			createdISBR := &apiv1.ISBServiceRollout{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, resourceLookupKey, createdISBR)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("Verifying the content of the ISBServiceRollout spec")
+			createdInterStepBufferServiceSpec := numaflowv1.InterStepBufferService{}
+			Expect(json.Unmarshal(createdISBR.Spec.InterStepBufferService.Raw, &createdInterStepBufferServiceSpec)).ToNot(HaveOccurred())
+			Expect(createdISBR.Spec).Should(Equal(isbServiceRollout.Spec))
+		})
```

Comment:
This seems to be repeated from L76-L85. You should instead verify ISBService is created, right?